### ISSUE TITLE
generator raised StopIteration

### ISSUE THREAD

**Issue by ishkania** [nescio007/teether#19]
Hi there,

I'm quite new to programming (still in high school) and I've been wanting to learn more about teEther. I've been running it on a few test contracts that I'm writing, but I sometimes get the following error. Can you please advise on the reason? 

Thank you very much, huge fan of your work!

-----------------------------------------------------------

File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/teether-0.1.8-py3.9.egg/teether/explorer/forward.py", line 73, in find
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/teether-master/bin/gen_exploit.py", line 107, in <module>
    main(config['code'], config['target-address'], config['shellcode-address'], config['target_amount'],
  File "/teether-master/bin/gen_exploit.py", line 47, in main
    result = combined_exploit(p, int(target_addr, 16), int(shellcode_addr, 16), amount, amount_check,
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/teether-0.1.8-py3.9.egg/teether/exploit.py", line 216, in combined_exploit
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/teether-0.1.8-py3.9.egg/teether/project.py", line 86, in get_constraints
RuntimeError: generator raised StopIteration

**Comment by ishkania** [comment]
Sorry, I forgot to add the following snippet to my previous post. The Traceback immediately before the File "/Library" bit is usually at this spot:

------------------------------------------------------------------

INFO:root:No DELEGATECALL instructions
INFO:root:No CALLCODE instructions
INFO:root:No SELFDESTRUCT instructions
INFO:root:Found 3 RETURN and STOP instructions
Traceback (most recent call last):
File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/teether-0.1.8-py3.9.egg/teether/explorer/forward.py", line 73, in find....

**Comment by nescio007** [comment]
Hi, 
thanks a lot for your report, that was something I had not known before!

The problem was the following: at the point of the exception, the `find` function is looking for paths (slices) from a `SSTORE` instructions to `RETURN` or `STOP` instructions. All slices should then be returned from a generator. However, in the case that no such paths could be found I had implemented the abortion of the generator wrong. In particular, I raised a `StopIteration` when I should have just returned instead.
Interestingly, this used to work in older versions of python. Only [python3.7](https://docs.python.org/3/whatsnew/3.7.html#changes-in-python-behavior) added a change ([PEP 479](https://www.python.org/dev/peps/pep-0479/)), that now transforms a `StopIterations` to a `RuntimeError`.

Commit 5de8257 should fix the behaviour.
